app/main.py

```diff
@@ -2,0 +1,58 @@
+class SoftwareEngineer:
+    skills = []
+
+    def __init__(self, name: str) -> None:
+        self.name = name
+        self.skills = []
+
+    def learn_skill(self, skill: str) -> None:
+        if skill not in self.skills:
+            self.skills.append(skill)
+
+
+class FrontendDeveloper(SoftwareEngineer):
+    skills = ["JavaScript", "HTML", "CSS"]
+
+    def __init__(self, name: str) -> None:
+        super().__init__(name)
+        self.skills = FrontendDeveloper.skills.copy()
```

Comment:
Using `FrontendDeveloper.skills.copy()` ensures that each instance gets its own copy of the skills list, which is good practice. However, the class variable `skills` in `SoftwareEngineer` might still cause confusion and should be removed.